### PR TITLE
retry failed elevation tile download

### PR DESCRIPTION
**Pull request by nilsnolde**
fixes https://github.com/valhalla/valhalla/issues/4425

### CODE REVIEW

File: scripts/valhalla_build_elevation [hunk 1]
```diff
@@ -237,39 +244,69 @@ def download(tile: Tile, output_dir, compression: TileCompression):
...
-                            f.write(compressor.compress(uncompressed))
-                            f.write(compressor.flush())
-
-            LOGGER.debug(f"Successfully downloaded tile {tile.name}")
-
-        return True
-    except URLError as e:
-        LOGGER.critical(f"Download failed of elevation tile {tile.dir}/{tile.name}: {e.reason}")
+
+    download_status = DownloadStatus.FAILED
```
Comment (nilsnolde) [inline scripts/valhalla_build_elevation:248]: most of the diff is just wrapping the previous logic into a for loop, where we request 15 times maximum with increased sleeps in between (exponential back off), up to 4.5 hours until we give up. could give much earlier though :D

File: scripts/valhalla_build_elevation [hunk 2]
```diff
@@ -237,39 +244,69 @@ def download(tile: Tile, output_dir, compression: TileCompression):
...
+            sleep((i**2) / 2 * 30)
+            with request.urlopen(url) as res, open(filepath, "wb") as f:
+
+                if compression is TileCompression.GZIP:
+                    f.write(res.read())
+                else:
+                    with gzip.GzipFile(fileobj=res, mode="rb") as gz:
+                        try:
+                            uncompressed = gz.read()
+                        except Exception as e:
```
Comment (nilsnolde) [inline scripts/valhalla_build_elevation:262]: protect for corrupt downloads

File: scripts/valhalla_build_elevation [hunk 3]
```diff
@@ -237,39 +244,69 @@ def download(tile: Tile, output_dir, compression: TileCompression):
...
+            continue
+        except ImportError:
+            LOGGER.critical(
+                "Could not import lz4. Please install lz4 or use another compression format."
+            )
+            sys.exit(1)
+
+    if download_status == DownloadStatus.CORRUPTED:
+        LOGGER.error(f"Tile {tile.dir}/{tile.name} was corrupted, removing it...")
+        filepath.unlink()
```
Comment (nilsnolde) [inline scripts/valhalla_build_elevation:303]: if it was corrupted in the end, let's remove that tile from the file system

**Review by kevinkreiser — APPROVED** [review]
maybe use 4 or something instead of 15 tries

**Comment by nilsnolde** [comment]
changed it to 5 retries, totalling 7.5 mins in worst case.

**Review by kevinkreiser — APPROVED** [review]

**Comment by nilsnolde** [comment]
not gonna merge master again with the admin changes

**Pull request merged**